samcli/commands/init/init_templates.py

```diff
@@ -102,6 +106,37 @@ def clone_templates_repo(self):
                 if expected_previous_clone_local_path.exists():
                     self._git_repo.local_path = expected_previous_clone_local_path
 
+    def _check_upsert_templates(self, shared_dir: Path, cloned_folder_name: str) -> bool:
+        """
+        Check if the app templates repository should be cloned, or if cloning should be skipped.
+
+        Parameters
+        ----------
+        shared_dir: Path
+            Folder containing the aws-sam-cli shared data
+
+        cloned_folder_name: str
+            Name of the directory into which the app templates will be copied
+
+        Returns
+        -------
+        bool
+            True if the cache should be updated, False otherwise
+
+        """
+        cache_dir = shared_dir / cloned_folder_name
+        git_executable = self._git_repo.get_git_executable()
+        command = [git_executable, "rev-parse", "--verify", "HEAD"]
+        try:
+            existing_hash = check_output(command, cwd=cache_dir, stderr=STDOUT).decode("utf-8").strip()
+        except CalledProcessError:
+            LOG.debug("Unable to check existing cache hash")
```

Comment:
I wonder if we can show more info in this log about the `CalledProcessError`. Is there any message from the `CalledProcessError` that we can wrap into the debug log. Could be a useful info to us even if we don't actually throw an error here.